commcare_connect/opportunity/api/views.py

```diff
@@ -43,3 +52,27 @@ class UserVisitViewSet(viewsets.GenericViewSet, viewsets.mixins.ListModelMixin):
 
     def get_queryset(self):
         return UserVisit.objects.filter(opportunity=self.kwargs.get("opportunity_id"), user=self.request.user)
+
+
+class ClaimOpportunityView(APIView):
+    authentication_classes = [OAuth2Authentication]
+    permission_classes = [TokenHasReadWriteScope]
+
+    def post(self, *args, **kwargs):
+        opportunity_access = get_object_or_404(OpportunityAccess, user=self.request.user, opportunity=kwargs.get("pk"))
+        opportunity = opportunity_access.opportunity
+        if opportunity.learn_app.cc_domain != opportunity.deliver_app.cc_domain:
+            create_hq_user(self.request.user, opportunity.deliver_app.cc_domain, opportunity.api_key)
```

Comment:
Could put this after the claim object creation so that we only call it if there wasn't already a claim?

Reply:
Alternately I guess we could re-call it as long as it's idempotent.